fix(api): update the plate reader serial number parser to include BYO and OPT delims.

vegano1 opened the pull request:
# Overview

The serial number format was updated to OPT instead of BYO, so let's update the parser to include both.

## Test Plan and Hands on Testing

- [x] Plate reader with `BYO` in serial number is shown on the desktop app with full serial number
- [ ] Plate reader with `OPT` in serial number is shown on the desktop app with full serial number

## Changelog

- Update the serial number parser for the plate reader to include both OPT and BYO delimiters.

## Review requests
## Risk assessment

low

vegano1 committed:
fix(api): update the plate reader serial number parser to include BYO and OPT delims.

nusrat813 reviewed (approved):
Tested, LGTM 
Old plate reader serial BYO

SyntaxColoring reviewed (approved):
Should we be using re.fullmatch (possibly combined with str.trim()) instead of re.match? Right now, if there are extra trailing digits, I think this will silently truncate them, which does not seem intentional given the way that the regex is written.

vegano1 committed:
serial number should match 1 or more digits
use fullmatch and strip to make sure we dont silently fail if the format is invalid
